lib/generateClass.js

```diff
@@ -549,6 +795,182 @@ function generateFHIRAssignment(cardIsList, baseCardIsList, constraintsLength, f
   });
 }
 
+/**
+ * Generates the "assignment" portion of the fromFHIR method, where the SHR elements get built from the FHIR json
+ * @param {boolean} cardIsList whether the element cardinality is a "list" (aka > 1)
+ * @param {boolean} baseCardIsList whether the element is "based" on an element whose cardinality is a list (> 1)
+ * @param {Number} constraintsLength the number of constraints on the element; used to determine whether there are more contstrained elements than cardinalities
+ * @param {string} fhirElement the FHIR element that the data should be taken from  (in FQN notation)
+ * @param {string} shrElementPath the SHR element that the data should be put into (in FQN notation)
+ * @param {Object} fhirProfile the FHIR profile the element comes from
+ * @param {Object} slicing information on this element related to slicing, if any
+ * @param {CodeWriter} cw the CodeWriter that is writing the file for this element
+ */
+function generateFromFHIRAssignment(cardIsList, baseCardIsList, constraintsLength, fhirElement, shrElementMapping, shrElementPath, fhirProfile, slicing, cw) {
+  const fhirElementPath = fhirElement.path;
+
+  const valueArray = fhirElementPath.split('.');
+  valueArray.shift(); // discard the first because it is the resource name
+  let prev = `fhir['${valueArray.shift()}']`; // start with this[firstInChain]
+  let check = `${prev} != null`; // e.g. this.firstInChain !== null && this.firstInChain.secondInChain !== null
+  valueArray.forEach(v => {
+    prev = `${prev}['${v}']`;
+    check = check + ' && ' + prev + ' != null';
+  });
+
+  // If we have more things (as evidenced by the 'IncludesTypeConstraints' that we need to fit into the end of the path)
+  // than we can fit into it, then we need to figure out where the closest array to the current element is.
+  // This will get put into 'containerProfile' if it exists.
+  let containerProfileArray;
+  if (!(cardIsList || baseCardIsList) && constraintsLength > 1) {
+    const pathArrayClone = fhirElementPath.split('.');
+    // loop over the pathArrayClone to figure out where th nearest 0..*-cardinality element is
+    while (pathArrayClone.length > 0) {
+      pathArrayClone.pop();
+      const searchString = pathArrayClone.join('.');
+      const candidateSnapshotElement = fhirProfile.snapshot.element.find(e => { return e.path === searchString; });
+      // If we find a snapshot element that matches the search string, and which has a 'max' (cardinality) of *
+      // That's our nearest array element (into which we're going to put our elements)
+      if (candidateSnapshotElement && candidateSnapshotElement.max == '*') {
```

Comment:
I think you also want to allow for cases where `max` is a number > 1 (but note it will be a stringified number).  For example, what if `max` is `"5"`?